src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +56,1180 @@ ActorInit Obj_Mine_InitVars = {
     sizeof(ObjMine),
     (ActorFunc)ObjMine_Init,
     (ActorFunc)ObjMine_Destroy,
-    (ActorFunc)ObjMine_Update,
-    (ActorFunc)ObjMine_Draw,
+    (ActorFunc)ObjMine_Path_Update,
+    (ActorFunc)ObjMine_Path_Draw,
 };
 
-// static ColliderJntSphElementInit sJntSphElementsInit[1] = {
-static ColliderJntSphElementInit D_80A84570[1] = {
+static ColliderJntSphElementInit sJntSphElementsInit[1] = {
     {
-        { ELEMTYPE_UNK2, { 0x00000000, 0x00, 0x00 }, { 0x01CBFFBE, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+        {
+            ELEMTYPE_UNK2,
+            { 0x00000000, 0x00, 0x00 },
+            { 0x01CBFFBE, 0x00, 0x00 },
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_ON,
+        },
         { 0, { { 0, 0, 0 }, 30 }, 100 },
     },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_80A84594 = {
-    { COLTYPE_METAL, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    ARRAY_COUNT(sJntSphElementsInit), D_80A84570, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_METAL,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
+    },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit,
+};
+
+static f32 sPathSpeeds[] = {
+    1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f,
+};
+
+static ObjMineMtxF3 sStandardBasis = {
```

Comment:
Both, since it's a rotation matrix.

Reply:
To be more specific, `x`, `y`, and `z` are the basis for the local coordinate system. Treated as a matrix, it is the rotation that converts vectors in the standard basis representation to the local basis representation. The matrix that gets put into the rotation is the transpose (AKA the inverse) of this, because it's converting a local basis representation to the standard basis representation.